tests/src/test/java/io/smallrye/concurrency/test/cdi/providedBeans/BeanWithInjectionPoints.java

```diff
@@ -0,0 +1,143 @@
+package io.smallrye.concurrency.test.cdi.providedBeans;
+
+import io.smallrye.concurrency.api.ManagedExecutorConfig;
+import io.smallrye.concurrency.api.NamedInstance;
+import io.smallrye.concurrency.api.ThreadContextConfig;
+import io.smallrye.concurrency.impl.ManagedExecutorImpl;
+import io.smallrye.concurrency.impl.ThreadContextImpl;
+import io.smallrye.concurrency.impl.ThreadContextProviderPlan;
+import io.vertx.core.cli.annotations.Name;
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ThreadContext;
+import org.eclipse.microprofile.concurrent.spi.ThreadContextProvider;
+import org.hibernate.engine.spi.Managed;
+import org.jboss.weld.proxy.WeldClientProxy;
+import org.junit.Assert;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * There are multiple context providers added in tests, we do not assert on those and only look for CDI and JTA now.
+ */
+@ApplicationScoped
+public class BeanWithInjectionPoints {
```

Comment:
Nice set of tests

Reply:
Thanks!
More coming for MP config usage in these injection points if I can find some time for that.